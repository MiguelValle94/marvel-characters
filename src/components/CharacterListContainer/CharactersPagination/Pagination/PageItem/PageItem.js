import React from 'react';
import PropTypes from 'prop-types';

function PageItem({ active, disabled, activeLabel, children, onClick }) {
  return (
    <li
      className={`page-item${disabled ? ' disabled' : ''}${
        active ? ' active' : ''
      }`}
    >
      <button className='page-link' disabled={disabled} onClick={onClick}>
        {children}
        {active && activeLabel && (
          <span className='sr-only'>{activeLabel}</span>
        )}
      </button>
    </li>
  );
}

PageItem.propTypes = {
  active: PropTypes.bool,
  disabled: PropTypes.bool,
  activeLabel: PropTypes.string,
  onClick: PropTypes.func
};

PageItem.defaultProps = {
  active: false,
  disabled: false,
  activeLabel: '(current)'
};

function createButton(name, defaultValue, label = name) {
  function Button({ children, ...props }) {
    return (
      <PageItem {...props}>
        <span aria-hidden='true'>{children || defaultValue}</span>
        <span className='sr-only'>{label}</span>
      </PageItem>
    );
  }

  Button.displayName = name;

  return Button;
}

export default PageItem;
export const First = createButton('First', '«');
export const Prev = createButton('Prev', '‹', 'Previous');
export const Ellipsis = createButton('Ellipsis', '…', 'More');
export const Next = createButton('Next', '›');
export const Last = createButton('Last', '»');
