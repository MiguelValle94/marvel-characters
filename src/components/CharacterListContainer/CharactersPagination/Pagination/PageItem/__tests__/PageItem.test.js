import React from 'react';
import { shallow } from 'enzyme';

import PageItem from '../PageItem';

const setUpShallow = ({ active, activeLabel, disabled, handleOnClick }) => {
  const component = shallow(
    <PageItem
      active={active}
      activeLabel={activeLabel}
      disabled={disabled}
      onClick={handleOnClick}
    >
      1
    </PageItem>
  );
  return component;
};

describe('PageItem', () => {
  let component;
  const handleOnClick = jest.fn();

  it('should display page item', () => {
    component = setUpShallow({ handleOnClick });
    expect(component).toMatchSnapshot();
  });

  it('should display page item - active', () => {
    component = setUpShallow({ active: true, activeLabel: 'active label' });
    const li = component.find('li');
    const span = component.find('span');
    expect(li.props().className.includes('active')).toBeTruthy();
    expect(span.props().children).toBe('active label');
  });

  it('should display page item - disabled', () => {
    component = setUpShallow({ disabled: true });
    const li = component.find('li');
    const button = component.find('button');
    expect(li.props().className.includes('disabled')).toBeTruthy();
    expect(button.props().disabled).toBe(true);
  });
});
