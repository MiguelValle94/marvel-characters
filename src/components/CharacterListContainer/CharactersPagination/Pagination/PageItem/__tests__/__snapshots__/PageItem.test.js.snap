// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageItem should display page item 1`] = `
<li
  className="page-item"
>
  <button
    className="page-link"
    disabled={false}
    onClick={[MockFunction]}
  >
    1
  </button>
</li>
`;
